### ISSUE TITLE
Add slugify function to std

### ISSUE THREAD

**Issue by marvinhagemeister** [denoland/std#5627]
**Is your feature request related to a problem? Please describe.**

A common part of building web apps includes generating pretty URLs based of a random string like a page title.

- `Hello World` -> `hello-world`
- `déjà vu` -> `deja-vu`
- `Foo & Bar` -> `foo-and-bar`
- ...etc

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Comment by kt3k** [comment]
Sounds like a good addition to `std/text`

**Comment by timreichen** [comment]
I can take that one. Looking over the implementation of https://www.npmjs.com/package/slugify, there are many options I think would probably not be necessary:
```ts
{
  replacement: '-',  // replace spaces with replacement character, defaults to `-`
  remove: undefined, // remove characters that match regex, defaults to `undefined`
  lower: false,      // convert to lower case, defaults to `false`
  strict: false,     // strip special characters except replacement, defaults to `false`
  locale: 'vi',      // language code of the locale to use
  trim: true         // trim leading and trailing replacement chars, defaults to `true`
}
```
- I think `strict`, `remove`, `lower`, `trim` should be done by default or that should be done manually.
- `replacement` seems like it should be `separator`. Does this vary with slugification `-`? Else instead of `slugify()` we could have a "remove special chars" function in combination with `@std/to*Case()` functions.

WDYT?

**Comment by kt3k** [comment]
I'm in favor of reducing number of options in first pass.

> I think strict, remove, lower, trim should be done by default

Do you mean we would implement it like all these options are true, or these set to the same as npm:slugify?

> replacement seems like it should be separator. Does this vary with slugification -

I don't think we need to make `separator` optional in first pass.

**Comment by kt3k** [comment]
I'd also suggest we would use `str.normalize("NFD").replace(/\p{Diacritic}/gu, "")` for removing diacritic characters instead of mapping with manually maintained list.

**Comment by timreichen** [comment]
> > I think strict, remove, lower, trim should be done by default
> 
> Do you mean we would implement it like all these options are true, or these set to the same as npm:slugify?

I am not too familiar with `slugify()`, I thought it should always be trimmed and lowercased. If so I propose to do it without an option. If not, it should be done like `slugify("....").trim().toLowerCase()`. Same with remove and strict.

**Comment by kt3k** [comment]
closed by #5646
